packages/console/src/console-history/SizeUsageUI.tsx

```diff
@@ -0,0 +1,211 @@
+          ) {
+            past.time.shift();
+            past.usage.shift();
+          }
+
+          past.time.push(time);
+          past.usage.push(usage);
+
+          historyUsage.current = past;
+        } else {
+          historyUsage.current = { time: [], usage: [] };
+        }
+
+        setUsage(newUsage);
+      };
+      fetchAndUpdate();
+
+      const updateUsage = setInterval(
+        fetchAndUpdate,
+        hover ? hoverUpdateInterval : defaultUpdateInterval
+      );
+      return () => {
+        clearInterval(updateUsage);
+      };
+    },
+    [
+      hover,
+      hoverUpdateInterval,
+      connection,
+      defaultUpdateInterval,
+      monitorDuration,
+      bgMonitoring,
+    ]
+  );
+
+  const toDecimalPlace = (num: number, dec: number) =>
+    Math.round(num * 10 ** dec) / 10 ** dec;
+
+  const decimalPlace = 2;
+  const GbToByte = 1024 ** 3;
+
+  const freeMemoryGB = toDecimalPlace(freeMemory / GbToByte, decimalPlace);
+  const totalHeapGB = toDecimalPlace(totalHeapSize / GbToByte, decimalPlace);
+  const maxHeapGB = toDecimalPlace(maxHeapSize / GbToByte, decimalPlace);
+  const inUseGB = totalHeapGB - freeMemoryGB;
+
+  const getRow = (text: string, size: string, bottomBorder = false) => (
+    <div
+      className={classNames(`heap-usage-info-row`, {
+        'heading-bottom-border': bottomBorder,
+      })}
+    >
+      <div>
+        <h6
+          style={{
+            fontWeight: 'bold',
+          }}
+        >
+          {text}
+        </h6>
+      </div>
+      <div style={{}}>
+        <h6>{size}</h6>
+      </div>
+    </div>
+  );
+
+  const timestamps = historyUsage.current.time;
+  const usages = historyUsage.current.usage;
+  const lastTimestamp = timestamps[timestamps.length - 1] ?? 0;
+  return (
+    <>
+      <div
+        className="max-memory"
+        onMouseEnter={() => setHover(true)}
+        onMouseLeave={() => setHover(false)}
+      >
+        <div
+          className="total-memory"
+          style={{ width: `${(totalHeapSize * 100) / maxHeapSize}%` }}
+        />
+        <div
+          className={classNames('used-memory', {
+            'heap-overflow': (totalHeapSize - freeMemory) / maxHeapSize > 0.95,
+          })}
+          style={{
+            width: `${((totalHeapSize - freeMemory) * 100) / maxHeapSize}%`,
+          }}
+        />
+
+        <div className="memory-text">{maxHeapGB.toFixed(1)} GB</div>
+      </div>
+
+      <Tooltip>
+        <div className="heap-tooltip">
+          {getRow(
+            'In use:',
+            `${inUseGB.toFixed(decimalPlace)} of ${maxHeapGB.toFixed(
+              decimalPlace
+            )} GB`,
+            true
+          )}
+          {getRow('Free:', `${freeMemoryGB.toFixed(decimalPlace)} GB`)}
+          {getRow('Total:', `${totalHeapGB.toFixed(decimalPlace)} GB`)}
+          {getRow('Max:', `${maxHeapGB.toFixed(decimalPlace)} GB`)}
+          <div className="heap-plot">
+            <Plot
```

Comment:
What is this plotting? Doesn't seem to make sense, it's going down as my usage is going up. Should be in use/max.

![image](https://user-images.githubusercontent.com/1576283/184698836-4ec08409-7309-4817-b0ff-f4e430d6df6e.png)